WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dynamiccard/DynamicCardsBuilder.kt

```diff
@@ -0,0 +1,67 @@
+package org.wordpress.android.ui.mysite.cards.dynamiccard
+
+import org.wordpress.android.fluxc.model.dashboard.CardModel
+import org.wordpress.android.fluxc.model.dashboard.CardModel.DynamicCardsModel.CardOrder
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.Dynamic
+import org.wordpress.android.ui.mysite.MySiteCardAndItemBuilderParams.DynamicCardsBuilderParams
+import org.wordpress.android.ui.utils.ListItemInteraction.Companion.create
+import javax.inject.Inject
+
+class DynamicCardsBuilder @Inject constructor() {
+    fun build(params: DynamicCardsBuilderParams, order: CardOrder): List<Dynamic>? {
+        if (!shouldBuildCard(params, order)) {
+            return null
+        }
+        return convertToDynamicCards(params, order)
+    }
+
+    private fun shouldBuildCard(params: DynamicCardsBuilderParams, order: CardOrder): Boolean {
+        if (params.dynamicCards != null && params.dynamicCards.pages.none { it.order == order }) return false
+        return true
+    }
+
+    private fun convertToDynamicCards(params: DynamicCardsBuilderParams, order: CardOrder): List<Dynamic> {
+        val cards = params.dynamicCards?.pages?.filter { it.order == order } ?: emptyList()
```

Comment:
The `pages` part on the model is due to copy pasting some code on the FluxC side. I took a note refactor this later 😅

Reply:
Tackled this separately in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2928